File: MyCookbook/RecipeMicroserviceAPI/Data/Entities/Recipe.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace RecipeMicroserviceAPI.Data.Entities
{
    public class Recipe : BaseEntity
    {
        [Required]
        [StringLength(255)]
        public string Name { get; set; }

        [Required]
        public long CookingTime { get; set; }

        [Required]
        public long Servings { get; set; }

        public bool IsPublic { get; set; }

        public DateTime Created { get; set; } = DateTime.Now;

        [Required]
        public long UserId { get; set; }

        [Required]
        public long RecipeCategoryId { get; set; }

        public RecipeCategory RecipeCategory { get; set; }

        public long SkillLevelId { get; set; }

        [Required]
        public SkillLevel SkillLevel { get; set; }

        public ICollection<PreparationStep> PreparationSteps { get; set; }

        public ICollection<RecipeIngredient> RecipeIngredients { get; set; }
    }
}

File: MyCookbookApp/src/app/shared/models/recipe.ts
export interface Recipe {
    id: number;
    name: string;
    cookingTime: number;
    servings: number;
    isPublic: boolean;
    isSaved: boolean;
    categoryName: string;
    skillLevel: string;
}

export class RecipeInsert{
    id: number;
    name: string;
    cookingTime: number;
    servings: number;
    isPublic: boolean;
    recipeCategoryId: number;
    skillLevelId: number;
    userId: number;

    preparationSteps: [{
        stepNumber: number;
        stepText: string;
    }];

    recipeIngredients: [{
        ingredientId: number;
        unitId: number;
        quantity: number;
    }];
}

export class RecipeModel implements Recipe{
    id: number;
    name: string;
    cookingTime: number;
    servings: number;
    isPublic: boolean;
    isSaved: boolean;
    categoryName: string;
    skillLevel: string;
    userId: number;

    preparationSteps: [{
        stepNumber: number;
        stepText: string;
    }];

    recipeIngredients: [{
        ingredient: string;
        unit: string;
        quantity: number;
    }];
}

export class CookbookRecipe {
    id: number;
    recipe: Recipe;
    userName: string;
    savedAt: Date;
}
